### PR TITLE
add the bat-helper server

### PR DESCRIPTION
**Pull request by mrose17**
to run service, configure `TOKEN_LIST`, `HELPER_GITHUB_CLIENT_ID` and `HELPER_GITHUB_CLIENT_SECRET`

to use service in `bat-ledger` or `bat-eyeshade`, configure `HELPER_URL` and `HELPER_TOKEN`

Fixes https://github.com/brave-intl/bat-ledger/issues/55

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -1,106 +1,132 @@
...
+
+  balance: {
+    portno: 3003,
+
+    f: () => {
+      if (process.env.LEDGER_URL) {
+        module.exports.ledger = { url : process.env.LEDGER_URL  || 'http://127.0.0.1:3001' }
+      }
+
+      helper()
```
Comment (evq) [inline config.js]: we still need uphold config if we are supporting the card based apis for older builds

File: docker-compose.yml [hunk 2]
```diff
@@ -87,3 +87,33 @@ services:
...
+      - "SERVICE=helper"
+    depends_on:
+      - mongo
+      - redis
+    networks:
+      - ledger
+  helper-web:
+    container_name: helper-web
+    ports:
+      - "3002:3002"
```
Comment (evq) [inline docker-compose.yml]: s/3002/3004

File: eyeshade/controllers/rates.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+   GET /v1/rates
+ */
+
+v1.read = { handler: (runtime) => {
+  return async (request, reply) => {
+    const result = { altrates: runtime.currency.altrates, fxrates: runtime.currency.fxrates, rates: runtime.currency.rates }
+    let validity
+
+    underscore.keys(result).forEach((key) => { if (underscore.keys(result[key]).length === 0) delete result[key] })
+    validity = Joi.validate(result, runtime.currency.schemas.rates)
```
Comment (evq) [inline eyeshade/controllers/rates.js]: is there a reason to explicitly validate here vs using the response validation below?

File: docker-compose.yml (hunk 2 above)
Comment (mrose17) [reply docker-compose.yml]: good catch!

File: eyeshade/controllers/rates.js (hunk 3 above)
Comment (mrose17) [reply eyeshade/controllers/rates.js]: another good catch! there's a hook that i can use.

File: config.js (hunk 1 above)
Comment (mrose17) [reply config.js]: good point.

**Comment by mrose17** [comment]
@evq - back to you... thanks for the thoughtful review!

**Pull request merged**